http/telegraf.go

```diff
@@ -455,13 +455,58 @@ var _ platform.TelegrafConfigStore = (*TelegrafService)(nil)
 
 // FindTelegrafConfigByID returns a single telegraf config by ID.
 func (s *TelegrafService) FindTelegrafConfigByID(ctx context.Context, id platform.ID) (*platform.TelegrafConfig, error) {
-	panic("not implemented")
+	var cfg platform.TelegrafConfig
+	err := s.client.get(path.Join(telegrafsPath, id.String())).
+		Header("Accept", "application/json").
+		DecodeJSON(&cfg).
+		Do(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &cfg, nil
 }
 
 // FindTelegrafConfigs returns a list of telegraf configs that match filter and the total count of matching telegraf configs.
 // Additional options provide pagination & sorting.
-func (s *TelegrafService) FindTelegrafConfigs(ctx context.Context, filter platform.TelegrafConfigFilter, opt ...platform.FindOptions) ([]*platform.TelegrafConfig, int, error) {
-	panic("not implemented")
+func (s *TelegrafService) FindTelegrafConfigs(ctx context.Context, f platform.TelegrafConfigFilter, opt ...platform.FindOptions) ([]*platform.TelegrafConfig, int, error) {
+	var queryPairs []queryPair
+	if f.OrgID != nil {
+		queryPairs = append(queryPairs, queryPair{
+			k: "orgID",
+			v: f.OrgID.String(),
+		})
+	}
+	if f.Organization != nil {
+		queryPairs = append(queryPairs, queryPair{
+			k: "organization",
+			v: *f.Organization,
+		})
+	}
+	if f.ResourceID != 0 {
+		queryPairs = append(queryPairs, queryPair{
+			k: "resourceID",
+			v: f.ResourceID.String(),
+		})
+	}
+	if f.UserID != 0 {
+		queryPairs = append(queryPairs, queryPair{
+			k: "userID",
+			v: f.UserID.String(),
+		})
+	}
+
+	var resp struct {
+		Configs []*platform.TelegrafConfig `json:"configurations"`
+	}
+	err := s.client.get(telegrafsPath).
```

Comment:
Is there a reason this isn't using the `traceClient` from [`client.go`](https://github.com/influxdata/influxdb/blob/14e7daa6e416d7d2fa0d4c7d83678cb0531ec587/http/client.go#L68) like the other services ([bucket](https://github.com/influxdata/influxdb/blob/5f19c6cace36ceeb68666e080bbd049381d2bef2/http/bucket_service.go#L764))

Reply:
it actually uses that piece, the difference here is the client is more about normalizing the http reqs/resp handling. The tokens/content type/yada yada are all baked in with simple means to override the defaults when necessary (maybe even functional option inputs 🤷‍♂). The client provides some helpers for making the common path super repeatable (like not forgetting status checks on resp and so forth).